src/successModal/index.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { Component } from 'react'
+import { canUseDOM, canUseEventListeners } from 'exenv'
+import { createPortal } from 'react-dom'
+import TransitionGroup from 'react-transition-group/TransitionGroup'
+import CustomTransition, { AnimationType } from 'transitions'
+import { color } from '_utils/branding'
+
+import cc from 'classcat'
+import Button from 'button'
+import style from './style'
+
+const KEYCODES = {
+  ESCAPE: 27,
+}
+
+export interface SuccessModalProps {
+  readonly isOpen?: boolean
+  readonly children?: JSX.Element | JSX.Element[]
+  readonly className?: Classcat.Class
+  readonly closeOnEsc?: boolean
+  readonly onConfirm: () => void
+  readonly confirmLabel?: string
+  readonly illustartionSrc: string
+  readonly illustartionText?: string
+  readonly large?: boolean
+}
+
+class SuccessModal extends Component<SuccessModalProps> {
+  private portalNode: HTMLElement
+
+  static defaultProps: Partial<SuccessModalProps> = {
+    isOpen: false,
+    closeOnEsc: false,
+    large: false,
+    illustartionText: '',
+  }
+
+  constructor(props: SuccessModalProps) {
+    super(props)
+    if (canUseDOM) {
+      this.portalNode = document.createElement('div')
+      document.body.appendChild(this.portalNode)
+    }
+  }
+
+  componentDidMount() {
```

Comment:
Is it copy / pasted from the modal ?

Reply:
Yes we have the same in other modal component.

Reply:
Can you inherit from it then? If this is the same component

Reply:
I finally reuse the Modal component to avoid copie/paste the logic.